src/main/java/routing/CustomHandlerRouter.java

```diff
@@ -0,0 +1,38 @@
+package com.td.HttpServer;
+import java.util.Collection;
+import java.util.ArrayList;
+public class CustomHandlerRouter implements IHandlerRouter {
+
+  private Collection<CustomRoute> routes;
+  private DefaultHandlerRouter defaultHandlerRouter;
+
+  public CustomHandlerRouter(DefaultHandlerRouter defaultHandlerRouter) {
+    this.defaultHandlerRouter = defaultHandlerRouter;
+    this.routes = new ArrayList<CustomRoute>();
+  }
+
+  public void addRoute(String method, String path, IHandler handler) {
+    CustomRoute newRoute = new CustomRoute(method, path, handler);
+    routes.add(newRoute);
+  }
+
+  public IHandler selectHandlerBadRequest() {
+    return new HandlerBadRequest();
+  }
+
+  public IHandler selectHandler(HttpRequest request) {
+    boolean found = false;
+    IHandler handler = selectHandlerBadRequest();
```

Comment:
Although this is initialized to badRequestHandler, it looks like it will always be overwritten by either the for loop or the by `if (found == false)` check after the for loop.

Maybe an integration test to expose this would be worthwhile?